### PR TITLE
Constant sampler: add option to always follow the parent's decision.

### LINKED ISSUE #180: Add a new constant sampler
Currently we support only two types of ConstantSampler: "ALWAYS_ON" and "ALWAYS_OFF" we should probably support a new one "ALWAYS_PARENT_DECISION" - which acts like "ALWAYS_OFF" if no parent.

### PR DESCRIPTION
**Pull request by g-easy**
Fixes #180.

### CODE REVIEW

File: src/opencensus/proto/trace/v1/trace_config.proto [hunk 1]
```diff
@@ -55,12 +55,22 @@ message ProbabilitySampler {
...
-  // Whether spans should be always sampled, or never sampled.
+  // DEPRECATED: whether spans should be always sampled, or never sampled.
   bool decision = 1;
+
+  // How spans should be sampled:
+  // - Always on
+  // - Always off
+  // - Always follow the parent Span's decision (off if no parent).
+  enum ConstantDecision {
+    ALWAYS_OFF = 0;
```
Comment (g-easy) [inline src/opencensus/proto/trace/v1/trace_config.proto]: What do you think about adding UNSET = 0;

This is for backwards compatibility: if decision_v2 is "unset" (i.e. zero default due to proto3), then decision_v1 takes precedence.

File: src/opencensus/proto/trace/v1/trace_config.proto [hunk 2]
```diff
@@ -55,12 +55,15 @@ message ProbabilitySampler {
...
-// on span sampling.
+// Sampler that always makes a constant decision on span sampling.
 message ConstantSampler {
 
   // Whether spans should be always sampled, or never sampled.
   bool decision = 1;
+
+  // Whether to follow the parent's decision. If there is no parent, use
+  // the decision specified above.
+  bool follow_parent = 2;
```
Comment (g-easy) [inline src/opencensus/proto/trace/v1/trace_config.proto]: How about this option: just add another bool.

Comment (bogdandrutu) [reply src/opencensus/proto/trace/v1/trace_config.proto]: I think we don't use this yet, what about we just break compatibility and do the enum version?

Comment (g-easy) [reply src/opencensus/proto/trace/v1/trace_config.proto]: I'm okay with that.

**Comment by g-easy** [comment]
PTAL.

**Pull request merged**